tools/install_tools.sh

```diff
@@ -0,0 +1,282 @@
+#!/bin/bash -e
+
+usage() {
+  echo "Usage: $0 [openpilot install directory]"
+  echo
+  echo "Options:"
+  echo " -h | --help - Shows basic about/usage"
+  echo
+  echo "If no install directory is provided, a best-guess is used."
+}
+
+# Validate syntax / parse args
+for argval in "$@"; do
+  if [ "$argval" == --help ] || [ "$argval" == -h ]; then
+    usage
+    exit 0
+  fi
+done
+if [[ ! "$1" =~ (^\/.+$|^~.*$|^\.$|^$) ]]; then
+  echo "[FAIL] Invalid syntax"
+  echo
+  usage
+  exit 1
+else
+  if [ -z "$1" ]; then
+    OP_HOME=$(git -C "$PWD" rev-parse --show-toplevel)
+    if [[ "$OP_HOME" == fatal* ]]; then
+      echo "[FAIL] OP_HOME: $OP_HOME"
+      echo
+      echo "Please first change to a directory within the openpilot repository "
+      echo "or specify a folder when invoking $0."
+      echo
+      echo "Git folder result: $OP_HOME"
+      echo
+      exit 1
+    fi
+  else
+    OP_HOME="$1"
+  fi
+fi
+GIT_REMOTE=$(git -C "$OP_HOME" remote get-url origin)
+GIT_ROOT=$(git -C "$OP_HOME" rev-parse --show-toplevel)
+
+PLATFORM=$(uname)
+check_platform() {
+  if [[ $PLATFORM == Linux ]] || [[ $PLATFORM == Darwin ]]; then
+    echo "[PASS] PLATFORM: $PLATFORM"
+  else
+    echo "[FAIL] PLATFORM: $PLATFORM"
+    echo
+    echo "Today, openpilot can only be set up on Mac & Linux. Tomorrow, with"
+    echo "help of your generous contribution, we may support more platforms."
+    echo 
+    exit 1
+  fi
+}
+check_platform
+
+check_shell() {
+  if [[ $SHELL == "/bin/zsh" ]]; then
+    RC_FILE="$HOME/.zshrc"
+  elif [[ $SHELL =~ .+\/bash$ ]]; then
+    RC_FILE="$HOME/.bash_profile"
+  else
+    echo "[FAIL] SHELL: $SHELL"
+    echo
+    echo "Error: Shell in use ($SHELL) is not compatible with the openpilot "
+    echo "environment (yet)"
+    echo
+    echo "If you wish to keep a different default shell than bash (not "
+    echo "explicitly supported), you can invoke the bash shell by simply "
+    echo "typing the command `bash` then rerunning this script."
+    exit 1
+  fi
+  echo "[PASS] SHELL: $SHELL"
+}
+
+check_op_home() {
+  if [[ ! "$GIT_REMOTE" =~ .+\/openpilot.git$ ]] || [[ ! "$GIT_ROOT" == "$OP_HOME" ]]; then
+    echo "[FAIL] OP_HOME: $OP_HOME"
+    echo
+    echo "$OP_HOME should be where openpilot is cloned."
+    echo
+    echo "Expected: origin url should end in openpilot.git"
+    echo "Actual: $GIT_REMOTE"
+    echo
+    echo "Expected: git top level directory (root) should equal $OP_HOME"
+    echo "Actual: $GIT_ROOT"
+    echo
+    exit 1
+  else
+    echo "[PASS] OP_HOME: $OP_HOME"
+  fi
+}
+
+check_shell
+check_op_home
+
+install_deps_apt() {
+  APT_VERSION=$(apt-get --version | awk -F "." '/apt/ {print $1}')
+  if [[ $APT_VERSION != "apt 2" ]]; then
```

Comment:
this seems pretty excessive. has anyone actually run into this?

Reply:
this was the more explicit form of checking for apt package manager as a whole. my intent was to make it fail on distros which don't use apt, while not penalizing non-ubuntu users so long as their system meets requirements.